### PR TITLE
Add Redoc documentation for REST API reference

### PR DESCRIPTION
**Pull request by zangell44**
This PR makes some improvements to our open source REST API docs, starting with using Redoc instead of Swagger.

Preview of [REST API docs](https://deploy-preview-7503--prefect-orion.netlify.app/api-ref/rest-api-reference/)

Preview of [REST API overview](https://deploy-preview-7503--prefect-orion.netlify.app/api-ref/rest-api/)

Note that the Redoc three-column format needs a little more horizontal space, so we moved the interactive API docs to a separate page that has navigation turned off, making the entire horizontal space available. A link is provided to return to the overview. 

Should the lack of left navigation prove problematic to users, we can reconsider this approach. I think the audience for this API documentation will be okay.

Useful links for future work:

- Redoc styling options https://redocly.com/docs/api-reference-docs/configuration/theming/

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [ ] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`

### CODE REVIEW

**Review by billpalombi — APPROVED** [review]
Wow this is nice.

FWIW, the REST API Reference page feels like a page within a page. ave you guys already considered giving redoc the full page beneath the nav bar?

**Comment by tpdorsey** [comment]
@billpalombi Having nothing but the Redoc docs on the page was a thought, and I'm still open to it. My concern was whether users could easily navigate away from the API reference without at least minimal contextual description and link. But maybe that's overthinking the use case here.

**Comment by billpalombi** [comment]
I totally defer to you on this. Just sharing my knee jerk reaction. If you've already tried it, thats good enough for me.

File: docs/api-ref/rest-api/schema.json [hunk 1]
```diff
@@ -0,0 +1 @@
...ess those new fields gracefully. However,\nwhen PREFECT_TEST_MODE is on, extra fields are forbidden in order to catch\nsubtle unintentional testing errors."}, "WorkQueueUpdate": {"title": "WorkQueueUpdate", "type": "object", "properties": {"description": {"title": "Description", "type": "string", "description": "An optional description for the work queue.", "default": ""}, "is_paused": {"title": "Is Paused", "type": "boolean", "description": "Whether or not the work queue is paused.", "default": false}, "concurrency_limit": {"title": "Concurrency Limit", "type": "integer", "description": "An optional concurrency limit for the work queue."}, "last_polled": {"title": "Last Polled", "type": "string", "description": "The last time an agent polled this queue for work.", "format": "date-time"}, "filter": {"title": "Filter", "allOf": [{"$ref": "#/components/schemas/QueueFilter"}], "description": "DEPRECATED: Filter criteria for the work queue.", "deprecated": true}, "name": {"title": "Name", "type": "string", "description": "DEPRECATED: The name of the work queue.", "deprecated": true}}, "additionalProperties": false, "description": "Data used by the Orion API to update a work queue."}}}}
```
Comment (madkinsz) [inline docs/api-ref/rest-api/schema.json:1]: I don't think these belong in version control, right?

**Comment by madkinsz** [comment]
This is really sweet!

I'm a bit confused about the double `schema.json` files — did those get committed by accident? Can we just create those at build time?

Strong agree with Bill regarding the "page within a page" feeling. It feels crunched for space still and disconnected stylistically. I found myself looking for a way to open it in a new dedicated window.

**Comment by tpdorsey** [comment]
Thanks for the feedback so far. 

I've reworked the introduction to put more emphasis on accessing the docs directly on the API, particularly for Cloud. This is actually the correct and currently only way to access the full Cloud API documentation, and will always be correct for the running Cloud API regardless of doc publishing cadence.

Likewise, added emphasis that the Orion API docs are available both via a running Orion server instance and published in the docs. These should be always in sync.

We are somewhat limited in what we can do in the reference page itself. I've pared this back to the minimal chrome. We can further explore whether templating a special page for these API docs is appropriate. I'm currently leaning toward keeping users within a clear docs context for the present.

**Comment by tpdorsey** [comment]
Sorry about that flurry of commits. Turns out we weren't building the redoc docs from the correct version of the schema. JavaScript paths make my head hurt.

**Comment by billpalombi** [comment]
Is there any way to remove the title from the "Prefect Orion REST API Reference" page such that redoc is directly beneath the header nav? That would go some way toward alleviating my "page within a page" concern.

**Comment by zangell44** [comment]
Looks great! (I can't approve though)
Only thing to note is the REST API docs looks a little funny in dark mode, a lot of the text is hard to see against the dark background.

**Comment by tpdorsey** [comment]
@billpalombi I believe creating a custom template override for this page would enable removing the page title. Will look into this.

@zangell44 found what appears to be dark mode theme, but I need to see if that's switchable, and if it can be switched to match the current mode of the rest of the docs.

**Comment by tpdorsey** [comment]
@billpalombi would you be okay with leaving the page title in for now, and we'll loop back to addressing this in future? I believe it can be done, but requires reconfiguration of templates for all of the docs. This is on the roadmap related to future API docs formatting work.

**Comment by billpalombi** [comment]
> @billpalombi would you be okay with leaving the page title in for now, and we'll loop back to addressing this in future? I believe it can be done, but requires reconfiguration of templates for all of the docs. This is on the roadmap related to future API docs formatting work.

Yes, that's fine. Thanks for the context.

**Comment by madkinsz** [comment]
Is this ready for release?

**Comment by tpdorsey** [comment]
@madkinsz yes we can release this

**Pull request merged**